camel/toolkits/search_toolkit.py

```diff
@@ -305,7 +305,86 @@ def query_wolfram_alpha(self, query: str, is_detailed: bool) -> str:
+            query (str): The query to be searched.
+            num_results (int): The number of search results to retrieve
+                (default is 10).
+            **kwargs: Additional optional parameters supported by Tavily's API:
+                - search_depth (str): "basic" or "advanced" search depth.
+                - topic (str): The search category, e.g., "general" or "news."
+                - days (int): Time frame in days for news-related searches.
+                - max_results (int): Max number of results to return
+                  (overrides `num_results`).
+                See https://docs.tavily.com/docs/python-sdk/tavily-search/
+                api-reference for details.
+
+        Returns:
+            List[Dict[str, Any]]: A list of dictionaries representing search
+            results. Each dictionary contains:
+                - 'result_id' (int): The result's index.
+                - 'title' (str): The title of the result.
+                - 'description' (str): A brief description of the result.
+                - 'long_description' (str): Detailed information, if available.
+                - 'url' (str): The URL of the result.
+                - 'content' (str): Relevant content from the search result.
+                - 'images' (list): A list of related images (if
+                  `include_images` is True).
+                - 'published_date' (str): Publication date for news topics
+                  (if available).
+
+        Example:
+            {
+                'result_id': 1,
+                'title': 'OpenAI',
+                'description': 'An AI research organization...',
+                'long_description': 'OpenAI is a non-profit...',
+                'url': 'https://www.openai.com',
+                'content': 'OpenAI focuses on AI safety...',
+                'images': ['https://image.url/1.jpg'],
+                'published_date': '2024-09-15'
+            }
+        """
+        from tavily import (
+            InvalidAPIKeyError,
+            MissingAPIKeyError,
+            TavilyClient,
+            UsageLimitExceededError,
+        )
+
+        Tavily_API_KEY = os.getenv("TAVILY_API_KEY")
+        if not Tavily_API_KEY:
+            raise ValueError(
+                "`TAVILY_API_KEY` not found in environment variables. "
+                "Get `TAVILY_API_KEY` here: `https://www.tavily.com/api/`."
+            )
+
+        client = TavilyClient(Tavily_API_KEY)
+
+        try:
+            results = client.search(query, max_results=num_results, **kwargs)
+            return results
+        except MissingAPIKeyError:
+            return [{"error": "Missing API key."}]
+        except InvalidAPIKeyError:
+            return [{"error": "Invalid API key."}]
+        except UsageLimitExceededError:
+            return [
+                {
+                    "error": "Usage limit exceeded. Check your plan's "
+                    "usage limits."
+                }
+            ]
```

Comment:
this is great to handle specific error, but from code concise and maintain's perspective, I would suggest just use 
```
        except Exception as e:
            return [{"error": f"An unexpected error occurred: {e!s}"}]
```
to catch all exceptions